gluonnlp/model/train/embedding.py

```diff
@@ -48,68 +55,55 @@ def __init__(self, embedding_size, **kwargs):
-        batch_size : int, default 1024
-            Use batches of `batch_size` to compute the embeddings from the
-            `embedding_model`.
-        ctx
-            Context to perform computation on.
-        """
+        token : str
+            A token.
 
-        new_embedding = emb.TokenEmbedding(unknown_token=None)
-        new_embedding._token_to_idx.update(
-            (token, idx) for idx, token in enumerate(tokens))
-        new_embedding._idx_to_token = tokens
-        new_idx_to_vec = []
-
-        # Compute embeddings in batches for all idx
-        start_pointer = 0
-        end_pointer = batch_size
-        while True:
-            batch = tokens[start_pointer:end_pointer]
-            batch_embedding = self._to_token_embedding_batch(
-                batch, unknown_behavior, ctx)
-            new_idx_to_vec.append(batch_embedding.as_in_context(cpu()))
+        Returns
+        -------
+        bool:
+            True if a vector for token can be computed.
+        """
+        raise NotImplementedError
 
-            if end_pointer >= len(tokens):
-                break
-            start_pointer += batch_size
-            end_pointer += batch_size
+    def __getitem__(self, tokens):
+        """Looks up embedding vectors of text tokens.
 
-        new_idx_to_vec = nd.concat(*new_idx_to_vec, dim=0)
-        new_embedding._idx_to_vec = new_idx_to_vec
-        return new_embedding
+        Parameters
+        ----------
+        tokens : str or list of strs
+            A token or a list of tokens.
+
+        Returns
+        -------
+        mxnet.ndarray.NDArray:
+            The embedding vector(s) of the token(s). According to numpy
+            conventions, if `tokens` is a string, returns a 1-D NDArray
+            (vector); if `tokens` is a list of strings, returns a 2-D NDArray
+            (matrix) of shape=(len(tokens), vec_len).
+        """
+        raise NotImplementedError
 
 
-class SimpleEmbeddingModel(EmbeddingModel, HybridBlock):
+class SimpleEmbeddingModel(EmbeddingModel, Block):
     """A trainable embedding model.
 
     This class is a simple wrapper around the mxnet.gluon.nn.Embedding. It
     trains independent embedding vectors for every token. It implements the
     `gluonnlp.model.train.EmbeddingModel` interface which provides convenient
-    functions.
+    helper methods.
 
     Parameters
     ----------
     token_to_idx : dict
         token_to_idx mapping of the vocabulary that this model is to be trained
-        with. token_to_idx is used when to_token_embedding is called. For
+        with. token_to_idx is used for __getitem__ and __contains__. For
         initialization len(token_to_idx) is used to specify the size of the
-        subword embedding matrix..
+        subword embedding matrix.
     embedding_size : int
         Dimension of embeddings.
     weight_initializer : mxnet.initializer.Initializer, optional
```

Comment:
Flattered to see the default value of sparse_grad to be `True`

Reply:
Thanks for your great work on the sparsity support for mxnet!